### PR TITLE
Animation documentation changes

### PR DESCRIPTION
**Pull request by czakdesign**
The animations content was revised to address:
#13932 -  update the landing page to introduce implicit and transition animations
#14017 - add code sample to the API docs

Click the Animations link in the left-side nav: [Animations](https://cz-flutter-docs-staging.firebaseapp.com/)

Please review:  @InMatrix @HansMuller @Sfshaza @galeyang

The updates reflect the following:
- Reviewed all animation docs currently in flutter.io
- Reorganized the content
- Revised the Animation home page
- Added new pages for Implicit, Transition, and Explicit animations
- Added tables listing the widgets in each of the 3 animation types with links to the API docs.
- Added animation samples - 2 each for Implicit and Transition (i'm still working on these) - we decided not to include a sample for each widget at this time in order to get the new pages published
- Added a Concepts page - to catch all of the conceptual information in one place
_ Added images/diagrams/gif files

__ I'm revising content in the Transition and Explicit animations pages -- but please provide review comments on the content that is already there. 

Shams, I'll upload the image source files when all content and image files are final.

Thanks,

Christine

### CODE REVIEW

**Comment by Sfshaza** [comment]
Thanks, Christine!

File: animations/index.md [hunk 1]
```diff
@@ -1,159 +1,88 @@
...
 
-If there is specific animation documentation you'd like to see, please
-[file an issue](https://github.com/flutter/flutter/issues).
+ *Implicit animations* are animations that are already programmed or styled within pre-existing constraints—all you need to do is add the animated object, and set some basic parameters such as the duration. You don’t need to write the animation code to add listeners or tickers for example, and you don't need to tell the app when to start—the animation simply changes from the old to the new value.  
+ For more information, see [Implicit animations](/animations/implicit_widgets.html).
 
+
+### Transition animation widgets
+
+*Transition animations* implement the abstract class `AnimatedWidget`. Using the `AnimatedWidget` class allows you to separate the widget code from the animation code in the `setState()` call and it doesn’t need to maintain a `State` object to hold the animation.  Transition animations are similar to implicit animations because many of the animation elements are predefined for you. Use the  `AnimatedWidget` class (instead of `addListener()` and `setState()`) to create a reusable transition animation widget.
```
Comment (Sfshaza) [inline animations/index.md]: Another place where an example could convey this more succinctly. The InkWell's splash, for example.

File: animations/index.md [hunk 2]
```diff
@@ -1,159 +1,88 @@
...
+
+### Animation and Motion widgets  
+For a list of Flutter SDK animation widgets, see the following:
+* [Implicit animation widgets](/animations/implicit_widgets.html)  
+* [Transition animation widgets](/animations/transitions_widgets.html)  
+* [Explicit animation widgets](/animations/explicit_widgets.html)  
+* [Animation and Motion Widgets Catalog](/widgets/animation/)  
+
+### Codelabs  
+Codelabs are a great way to learn how to use Flutter and how to add Flutter animation widgets to your app. Check out [Building Beautiful UIs with Flutter](/codelabs/). This Codelab demonstrates how to build a simple chat app. And Step 7 (Animate your app) shows how to animate the new message—sliding it from the input area up to the message list. For more information, see [Codelabs](/codelabs/).
```
Comment (Sfshaza) [inline animations/index.md]: 1. Codelab => codelab
2. Delete the "For more info" sentence as you already link to Building Beautiful UIs and you haven't talked about the next codelabs yet.

File: animations/index.md [hunk 3]
```diff
@@ -1,159 +1,88 @@
...
+### Animation and Motion widgets  
+For a list of Flutter SDK animation widgets, see the following:
+* [Implicit animation widgets](/animations/implicit_widgets.html)  
+* [Transition animation widgets](/animations/transitions_widgets.html)  
+* [Explicit animation widgets](/animations/explicit_widgets.html)  
+* [Animation and Motion Widgets Catalog](/widgets/animation/)  
+
+### Codelabs  
+Codelabs are a great way to learn how to use Flutter and how to add Flutter animation widgets to your app. Check out [Building Beautiful UIs with Flutter](/codelabs/). This Codelab demonstrates how to build a simple chat app. And Step 7 (Animate your app) shows how to animate the new message—sliding it from the input area up to the message list. For more information, see [Codelabs](/codelabs/).  
+If you're new to Flutter, check out the [Write your first app, part 1](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt1/#0) and [Write your first app, part 2](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt2/#0) codelabs. They walk you through the basics of using Flutter to quickly create an iOS and Android app.
```
Comment (Sfshaza) [inline animations/index.md]: They walk... => Though they don't include animation code, they walk you...

File: animations/index.md [hunk 4]
```diff
@@ -1,159 +1,88 @@
...
+
+### Codelabs  
+Codelabs are a great way to learn how to use Flutter and how to add Flutter animation widgets to your app. Check out [Building Beautiful UIs with Flutter](/codelabs/). This Codelab demonstrates how to build a simple chat app. And Step 7 (Animate your app) shows how to animate the new message—sliding it from the input area up to the message list. For more information, see [Codelabs](/codelabs/).  
+If you're new to Flutter, check out the [Write your first app, part 1](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt1/#0) and [Write your first app, part 2](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt2/#0) codelabs. They walk you through the basics of using Flutter to quickly create an iOS and Android app.  
+
+### Flutter animation demos and examples  
+Flutter animation demos and examples provide code samples of some of the common and most popular animation widgets.  
+* [AnimatedList](/catalog/samples/animated-list/)—This example demonstrates how to display a list of cards which stay in sync using the  ListModel widget. When an item is added or removed from the model, the corresponding card animates in or out of view.
```
Comment (Sfshaza) [inline animations/index.md]: 1. [AnimatedList example](/catalog..)
2. This example... => How to display a ...
3. which => that

File: animations/index.md [hunk 5]
```diff
@@ -1,159 +1,88 @@
...
+* [Hero Animations](/animations/hero-animations/)—This demo shows how to build standard hero animations, and hero animations that transform the image from a circular shape to a square shape during flight. The [Shrine Demo](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/shrine_demo.dart) shows another example showing hero animations.
```
Comment (Sfshaza) [inline animations/index.md]: 1. [Hero Animations](...) guide. How to build hero animations (also called shared element animations). The guide shows two examples: a standard hero animation, and one that transforms the image ....

File: animations/index.md [hunk 6]
```diff
@@ -1,159 +1,88 @@
...
+* [Flutter Gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery)—The Flutter Gallery app shows material design widgets and other material and animation features provided by the Flutter SDK.
```
Comment (Sfshaza) [inline animations/index.md]: 1. Move this to the end of the list.
2. Try to avoid using the same word twice in a sentence (Flutter).
3. => ... provided by the animation libraries.
4. "The Flutter Gallery app shows" => A one-stop demo app that shows off many of the Material Design widgets, and many types of animations.

File: animations/index.md [hunk 7]
```diff
@@ -1,159 +1,88 @@
...
+
+### Flutter animation demos and examples  
+Flutter animation demos and examples provide code samples of some of the common and most popular animation widgets.  
+* [AnimatedList](/catalog/samples/animated-list/)—This example demonstrates how to display a list of cards which stay in sync using the  ListModel widget. When an item is added or removed from the model, the corresponding card animates in or out of view.
+* [Hero Animations](/animations/hero-animations/)—This demo shows how to build standard hero animations, and hero animations that transform the image from a circular shape to a square shape during flight. The [Shrine Demo](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/shrine_demo.dart) shows another example showing hero animations.
+* [Flutter Gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery)—The Flutter Gallery app shows material design widgets and other material and animation features provided by the Flutter SDK.  
+* [Staggered Animations Demo](https://flutter.io/animations/staggered-animations/)—This demo shows how to build a staggered animation using Flutter.
```
Comment (Sfshaza) [inline animations/index.md]: 1. Use a relative URL, not an absolute.
2. This demo shows => How to build a staggered animation where elements start at different points on the timeline.

File: animations/transition_widgets.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
+---
+layout: page
+title: "Flutter transition animations"
+permalink: /animations/transitions_widgets.html
+---
+
+* TOC Placeholder
+{:toc}
+
+Transition animations implement the abstract class `AnimatedWidget`. The `AnimatedWidget` class allows you to separate the widget code from the animation code in the `setState()` call. The `AnimatedWidget` class doesn’t need to maintain a `State` object to hold the animation. Use the  `AnimatedWidget` helper class (instead of `addListener()` and `setState()`) to create a reusable transition animation widget.
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: This page begins conceptually. It's a bit much to get one's head around. Start with an English description of a Transition. Mention that they are special objects that aren't part of the animation library, etc.

File: animations/transition_widgets.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+## Using a transition animation widget
+
+<div>
+<table class="table" width="100%">
+  <col width="35%">
+  <col width="65%">
+	<tbody>
+    <tr>
+      <td><img src="images/transitions.png" alt="Transition animation widgets"></td>
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: 1. I like the idea of this list, but make it more of a general description without so much implementation detail.
2. Then link pretty quickly to an example for those who just want to see code.

File: animations/transition_widgets.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+      4. Add the animation object, the duration of the animation, and any parameters relating to the _transition_ animation. For example, you would set the < example > for a <code>animationexample</code> widget.
+      </td>
+    </tr>
+   </tbody>
+  </table>
+</div>
+
+To create your own reusable transition animations, create a widget that extends `AnimatedWidget`.
+
+For more complex or customized animations, you can build [Explicit animations](/animations/explicit_widgets.html) using the  `AnimationController` class.
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: 1. Delete the complex word again.  :)

File: animations/transition_widgets.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+   </tbody>
+  </table>
+</div>
+
+To create your own reusable transition animations, create a widget that extends `AnimatedWidget`.
+
+For more complex or customized animations, you can build [Explicit animations](/animations/explicit_widgets.html) using the  `AnimationController` class.  
+
+## Transition animation examples  
+The `animationexample` and `animationexample` are two transition widgets included in the Flutter SDK.
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: 1. Those examples aren't part of the SDK, are they? Aren't they examples that use widgets in the SDK?  (Where is a link to them?)
2. Be specific. The transition widgets are mostly in the Widgets class, with some in the MaterialWidgets class. They aren't part of the animation package.

File: animations/transition_widgets.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+    <a href="" onMouseOver="document.MyImage1.src='images/AnimatedOpacity_blue.gif';" onMouseOut="document.MyImage1.src='images/placeholder_image.png';">
+    <img src="images/placeholder_image.png" name="MyImage1" height="180" width="160" name="MyImage1">
+    </a></td>
+    <td>
+    The <code>animationexample</code> transition widget automatically < description > .<br>Mouseover the image to view the animation.
+    </td>
+  </tr>
+</table>
+
+<!-- <a href="" onMouseOver="document.MyImage1.src='images/AnimatedOpacity_blue.gif';" onMouseOut="document.MyImage1.src='images/AnimatedOpacity_blue.png';">
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: Christine, what an excellent addition to our docs! Thanks!

File: animations/transition_widgets.md [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+<img src="images/AnimatedOpacity_blue.png" name="MyImage1" height="180" width="160" name="MyImage1">
+</a>
+Mouseover the image to view the animation using the <code>AnimatedOpacity</code> implicit widget. -->
+
+<!-- <div>
+<img src="images/AnimatedOpacity_blue.png" name="img" height="180" width="160">
+
+ <script>
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: 1. File a bug to add this feature to the website, so that it's not buried down in content pages.

File: animations/transition_widgets.md [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+ </script>
+
+ <a href="#C4" onClick="document.img.src=imgOn2.src;">▶︎</a>
+ <a href="#C4" onClick="document.img.src=imgOff2.src;">◼︎</a>
+
+ This example shows an animation using the <code>AnimatedOpacity</code> implicit widget.
+</div> -->
+
+ <!-- <br>
+ The `AnimatedOpacity` implicit widget automatically transitions a child's opacity over a given duration whenever the given opacity changes. Notice that the listeners, tickers, and other animation elements are *implied* so you don't need to add them. The `AnimatedOpacity` class extends `ImplicitlyAnimatedWidget` and those elements are already included in the `ImplicitlyAnimatedWidget` class. -->
```
Comment (Sfshaza) [inline animations/transition_widgets.md]: Are those classes even used in the background? I don't know.

File: animations/concepts.md [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+permalink: /animations/concepts.html
+---
+
+* TOC Placeholder
+{:toc}
+
+
+This section describes the basic concepts behind the Flutter animation architecture.
+
+If you use Flutter's built-in implicit and transition widgets, you may not need to interact with all of the Flutter core animation concepts directly because most of the work required to code the animation is already done. But for more complex explicit animations, it's important to understand the animation concepts used to fine-tune and control your animations.
```
Comment (Sfshaza) [inline animations/concepts.md]: 1. I'd reword this to something like:

For implicit or transition animations, you won't need to understand all of these concepts, but if you build explicit animations, you need to understand how Flutter's animation mechanism works.

File: animations/concepts.md [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+
+## How it works
+Animation is the movement of an object or a change over time—like a fade in or out, or a bounce. An animation represents a value of a specific type that can change over time—specifically over the lifetime of the animation. Most animation widgets receive an `Animation` object as a parameter, from which the animation widget reads the current value of the animation and listens for changes to that value. It listens for "state" changes.
+
+Whenever the animation’s object value changes, the animation object notifies all the listeners (that were added with `addListener`). Typically, a `State` object that listens to an animation will call `setState` in its listener callback to notify the widget system that it needs to rebuild with the new value of the animation.
```
Comment (Sfshaza) [inline animations/concepts.md]: 1. I think a State object ALWAYS calls setState, doesn't it? Remove "Typically,"
2. will call => calls
3. This is a super important point.

File: animations/concepts.md [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+
+> insert image/representation of Animation>listerners>state>build>implicit/transition/explicit
```
[the remark this answers is not in the document]
Comment (Sfshaza) [reply animations/concepts.md]: Your diagrams are beautiful, by the way. I know you said you didn't include the source for them yet, which is fine, but don't forget to put them in _diagrams.  :D

File: animations/concepts.md [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+
+There are three Flutter SDK widgets that help animation widgets rebuild when a state change occurs.
```
Comment (Sfshaza) [inline animations/concepts.md]: 1. "There are three Flutter SDK widgets that..." => Three classes from the Widget library help rebuild the relevant part of the widget tree when a state change occurs.

File: animations/concepts.md [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+![Flutter animations](images/code_complexity4.png)
+
+* `ImplicitlyAnimatedWidget` is an abstract widget used to build implicit widgets that gradually change their values over a period of time. This widget is already included in Flutter SDK implicit animations so you don't need to add it to your animation unless you are building a reusable implicit animation widget.
```
Comment (Sfshaza) [inline animations/concepts.md]: Remove all these Flutter SDK usages.

File: animations/concepts.md [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+* Animations are objects with a time dependent value.  
+* The value is updated each time Flutter generates a frame.  
+* Animations are Listenables: they notify listeners when their value changes.  
+* A `StatefulWidget` listens to an animation and responds to notifications by calling `setState()`.  
+* Flutter animations are stateful widgets whose build method depends on one or more animation values.  
+* The `Animation<T>` class is an abstract class.  
+
+
+## Essential background concepts  
+Flutter animation is built upon basic Flutter and Dart concepts such as the `StatefulWidget` and the `StatelessWidget` and working with the `main.dart` and `pubspec.yaml` files. Be sure to review the information in [A Tour of the Flutter Widget Framework](https://flutter.io/widgets-intro/) for important basic information you'll need to build your app. And if you haven't already followed the introductory codelabs, you might want to start with [Write Your First Flutter App, part 1](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt1/#0) and [Write Your First flutter App, part 2](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt2/#0).
```
Comment (Sfshaza) [inline animations/concepts.md]: 1. Use a relative URL.

**Comment by Sfshaza** [comment]
Please fix the PR so that it doesn't include the 50+ files in Flutter for React Native. Ask Filip if you need help.

File: animations/implicit_widgets.md [hunk 21]
```diff
@@ -0,0 +1,334 @@
...
+ img2Off.src = "images/AnimatedCrossFade_orig.png";
+ </script>
+
+ <a href="#" onmouseover="document.img2.src=img2On.src;"><img src="images/AnimatedCrossFade_orig.png" name="img2" height="180" width="160"></a>
+ <!-- <a href="#" onmouseout="document.img2.src=img2Off.src;">◼︎</a> -->
+
+
+ The code for the `AnimatedCrossFade` example is shown below. The `AnimatedCrossFade` class extends `ImplicitlyAnimatedWidget` which already includes the listeners, tickers, and other elements that define the animation. The listeners, tickers, and other animation elements are *implied* so you don't need to add them.  
+
+{% prettify dart %}
```
Comment (Sfshaza) [inline animations/implicit_widgets.md]: 1. "prettify dart" AND "```Dart" are resulting in the ``` showing in the code example.

File: animations/index.md [hunk 22]
```diff
@@ -1,159 +1,88 @@
...
+* [Implicit animation widgets](/animations/implicit_widgets.html)  
+* [Transition animation widgets](/animations/transitions_widgets.html)  
+* [Explicit animation widgets](/animations/explicit_widgets.html)  
+* [Animation and Motion Widgets Catalog](/widgets/animation/)  
+
+### Codelabs  
+Codelabs are a great way to learn how to use Flutter and how to add Flutter animation widgets to your app. Check out [Building Beautiful UIs with Flutter](/codelabs/). This Codelab demonstrates how to build a simple chat app. And Step 7 (Animate your app) shows how to animate the new message—sliding it from the input area up to the message list. For more information, see [Codelabs](/codelabs/).  
+If you're new to Flutter, check out the [Write your first app, part 1](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt1/#0) and [Write your first app, part 2](https://codelabs.developers.google.com/codelabs/first-flutter-app-pt2/#0) codelabs. They walk you through the basics of using Flutter to quickly create an iOS and Android app.  
+
+### Flutter animation demos and examples
```
Comment (InMatrix) [inline animations/index.md]: The items below are not really "demos and examples," except the AnimatedList sample. In the current Animations landing page, we call them "Common animation patterns". Which I think is still a useful notion.

File: animations/index.md (hunk 5 above)
Comment (InMatrix) [reply animations/index.md]: Please use "Shared element transition" to refer to this animation pattern. "Hero" is the widget used to implement this kind of animation, and it's a term specific to Flutter. In contrast, "Shared Element Transitions" are known to Android developers as well (e.g., https://android-developers.googleblog.com/2018/02/continuous-shared-element-transitions.html).

File: animations/index.md [hunk 23]
```diff
@@ -1,159 +1,88 @@
...
-come with the standard motion effects defined in their design spec,
-but it's also possible to customize these effects.
-
-The following resources are a good place to start learning the Flutter
-animation framework. Each of these documents shows, step by step, how
-to write animation code.
-{% comment %}
-More documentation is in the works on how to implement common design
-patterns, such as shared element transitions,
-and physics-based animations.
```
Comment (InMatrix) [inline animations/index.md]: I'd like to see physics-based animations mentioned somewhere in this doc. Even though we might not have additional content for it for a while, we need to make sure people know it's possible to create physics-based animations. This is especially important if the developer wants to mimic certain iOS-specific animations.

File: animations/index.md [hunk 24]
```diff
@@ -1,159 +1,88 @@
...
-link out.
-* The [animation
-library](https://docs.flutter.io/flutter/animation/animation-library.html)
-in the [Flutter API documentation](https://docs.flutter.io/)<br>
-The animation API for the Flutter framework.
-{% endcomment %}
+The Flutter SDK animation widgets form the basis for rendering your animation on iOS and Android platforms. Flutter animations widgets are *implicit*, *transition*, or *explicit* animations. If you’re starting out in animations, or you want to quickly add an animation to your app, try out the implicit or transition animation widgets.
 
-<hr>
+![Flutter animations](images/code_complexity3.png)
```
Comment (InMatrix) [inline animations/index.md]: What does “Build” mean as the label for the Y axis? I would label the Y axis as “flexibility and expressiveness” and the X axis as "implementation complexity." As mentioned above, it would be great to have an accompanying table to lay out how these two dimensions change as the developer moves from one type of animation implementation to another.

File: animations/implicit_widgets.md [hunk 25]
```diff
@@ -0,0 +1,334 @@
...
+layout: page
+title: "Flutter implicit animations"
+permalink: /animations/implicit_widgets.html
+---
+
+* TOC Placeholder
+{:toc}
+
+
+If you’re new to Flutter or new to animation, consider starting with the built-in Flutter implicit widgets. Implicit animations are useful when you want to quickly add an animation without having to write a lot of animation code because the Flutter SDK implicit widgets already include the `AnimationController`, the ticker provider, the listener, and other animations elements.
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: How about "Implicit animations are useful when you want to quickly add ~~an animation~~ a common motion effect to your app." I'd delete the rest of the sentence, for two reasons 1) it discourages the user to explore the other two implementation approaches, and 2) it brings up a number of concepts the user doesn't need to know at this point.

File: animations/implicit_widgets.md [hunk 26]
```diff
@@ -0,0 +1,334 @@
...
+
+Implicit animations are distinct from explicit animations which involve building the animation action using `AnimationController`. And implicit animations are different from transition widgets since you don't need to create an animation widget using `AnimatedWidget`, you simply use an existing implicit animation widget such as `AnimatedOpacity` or `AnimatedIcon`.
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: The second sentence conflates the experience of _creating_ a transition widget from _using_ one. Using a transition widget does not involve `AnimatedWidget`. I think this paragraph doesn't need to talk about how to create a custom transition widget. 

In addition, it's probably helpful to clarify that instead of using an `AnimationController` explicitly, an implicit animation simply responds to changes of a specific property and make such changes appear smoothly.

File: animations/implicit_widgets.md [hunk 27]
```diff
@@ -0,0 +1,334 @@
...
+<table class="table" width="100%">
+  <col width="35%">
+  <col width="65%">
+	<tbody>
+    <tr>
+      <td><img src="images/implicit.png" alt="Implicit animation widgets"></td>
+      <td>
+      1. Create your <code>StatefulWidget</code>.<br>
+      2. Create the <code>State</code> class to hold the animation object and the implicit animation widget.<br>
+      3. Specify the <code>BuildContext</code> to return the implicit animation widget, such as <code>AnimatedOpacity</code>.<br>
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: These three steps can probably be condensed into "placing your implicit animation widget in a stateful widget's build method," since they are just the standard way of creating a stateful widget.

File: animations/implicit_widgets.md [hunk 28]
```diff
@@ -0,0 +1,334 @@
...
+      4. Add the animation object, the duration of the animation, and any parameters relating to the implicit animation. For example, you would set the opacity level for an implicit <code>AnimatedOpacity</code> widget.
+      </td>
+    </tr>
+   </tbody>
+  </table>
+</div>
+
+To create your own reusable implicit animations, create a widget that extends `ImplicitlyAnimatedWidget`.
+
+For more complex or customized animations, you can use a subclass of `AnimatedWidget` to create [Transition animations](/animations/transitions_widgets.html) such as the `DecoratedBoxTransition` or you can build your own [Explicit animations](/animations/explicit_widgets.html) using the `AnimationController`.
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: As mentioned earlier, `AnimatedWidget` is for _creating_ your own transition widget, not about _using_ existing ones provided by Flutter such as FadeTransition. 

In addition, it would be more useful to spell out what you mean by "more complex or customized animations" to help users decide if they really need to consider transition animations or even explicit animations.

File: animations/implicit_widgets.md [hunk 29]
```diff
@@ -0,0 +1,334 @@
...
+   </tbody>
+  </table>
+</div>
+
+To create your own reusable implicit animations, create a widget that extends `ImplicitlyAnimatedWidget`.
+
+For more complex or customized animations, you can use a subclass of `AnimatedWidget` to create [Transition animations](/animations/transitions_widgets.html) such as the `DecoratedBoxTransition` or you can build your own [Explicit animations](/animations/explicit_widgets.html) using the `AnimationController`.  
+
+## Implicit animation examples  
+The `AnimatedOpacity` and `AnimatedCrossFade` are two implicit widgets included in the Flutter SDK.
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: It's worth mentioning that there are many implicit animation widgets, but we use these two as examples.

File: animations/implicit_widgets.md [hunk 30]
```diff
@@ -0,0 +1,334 @@
...
+
+## Implicit animation examples  
+The `AnimatedOpacity` and `AnimatedCrossFade` are two implicit widgets included in the Flutter SDK.  
+
+### AnimatedOpacity example
+<!-- The `AnimatedOpacity` implicit widget automatically transitions a child's opacity over a given duration whenever the specified opacity changes. -->
+<table cellpadding="10">
+  <tr>
+    <td style="width:20%">
+    <a href="" onMouseOver="document.MyImage1.src='images/AnimatedOpacity_blue.gif';" onMouseOut="document.MyImage1.src='images/AnimatedOpacity_blue.png';">
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: The gif doesn't seem to be playing in my browser.

File: animations/implicit_widgets.md [hunk 31]
```diff
@@ -0,0 +1,334 @@
...
+ This example shows an animation using the <code>AnimatedOpacity</code> implicit widget.
+</div> -->
+
+ <!-- <br>
+ The `AnimatedOpacity` implicit widget automatically transitions a child's opacity over a given duration whenever the given opacity changes. Notice that the listeners, tickers, and other animation elements are *implied* so you don't need to add them. The `AnimatedOpacity` class extends `ImplicitlyAnimatedWidget` and those elements are already included in the `ImplicitlyAnimatedWidget` class. -->
+
+The code for the `AnimatedOpacity` example is shown below. The `AnimatedOpacity` widget extends the `ImplicitlyAnimatedWidget` class and already includes the listeners, tickers, and other elements that define the animation.  
+<!-- skip -->
+{% prettify dart %}
+```Dart
```
Comment (InMatrix) [inline animations/implicit_widgets.md]: > \`\`\`Dart 

This seems to be extranous. It showed up on the page.

File: animations/transition_widgets.md (hunk 8 above)
Comment (InMatrix) [inline animations/transition_widgets.md]: Can we talk about how to create your own transition widgets _after_ talking about how to use the transition widgets provided by Flutter? Most people probably don't need to create their own transition widgets.

File: animations/transition_widgets.md [hunk 32]
```diff
@@ -0,0 +1,200 @@
...
+  <col width="35%">
+  <col width="65%">
+	<tbody>
+    <tr>
+      <td><img src="images/transitions.png" alt="Transition animation widgets"></td>
+      <td> < Revise for a transition widget example >   
+      1. Create your <code>StatefulWidget</code>.<br>
+      2. Create the <code>State</code> class to hold the animation object and the _transition_  animation widget.<br>
+      3. Specify the <code>BuildContext</code> to return the _transition_ animation widget, such as <code>animationexample</code>.<br>
+      4. Add the animation object, the duration of the animation, and any parameters relating to the _transition_ animation. For example, you would set the < example > for a <code>animationexample</code> widget.
```
Comment (InMatrix) [inline animations/transition_widgets.md]: As I mentioned in the implicit animation doc, steps 1-3 is just how you create any stateful widget.

File: animations/transition_widgets.md [hunk 33]
```diff
@@ -0,0 +1,200 @@
...
+import 'package:flutter/material.dart';
+
+add the code for the transition widget
+
+```
+{% endprettify %}
+
+
+## Transition animations widgets
+
```
Comment (InMatrix) [inline animations/transition_widgets.md]: This is a useful list!

File: animations/explicit_widgets.md [hunk 34]
```diff
@@ -0,0 +1,166 @@
...
+add the code for the explicit widget
+
+```
+{% endprettify %}
+
+
+
+## Explicit animations widgets
+
+The Flutter SDK includes the following explicit animation widgets. This is a sample of the many explicit widgets that are included in the Flutter SDK.
```
Comment (InMatrix) [inline animations/explicit_widgets.md]: The widgets in the list below are widgets _using_ explicit animations in their implementations, not widgets for _building_ explicit animations. Using the widgets in the list require no knowledge about building animations. Since this doc is about creating explicit animations using AnimatedBuilder, I don't feel this list is relevant, unless the intent is to point the user to the source code of these widgets as references for writing explicit animations.

File: animations/concepts.md [hunk 35]
```diff
@@ -0,0 +1,298 @@
+---
+layout: page
+title: "Flutter animation concepts"
```
Comment (InMatrix) [inline animations/concepts.md]: I wish there were diagrams in the concepts doc to help illustrate how different concepts are related to one another, a pain-point both study participants and internal users reported. Without visual explanations, this doc doesn't look very different from clicking into each class's API doc and read the descriptions there. I made an attempt at this a while ago, please take a look: https://github.com/flutter/flutter/issues/13931#issuecomment-367849618 I'd be happy to work with you on adding more visual explanations.

File: animations/explicit_widgets.md [hunk 36]
```diff
@@ -0,0 +1,114 @@
+---
+layout: page
+title: "Flutter explicit animations"
+permalink: /animations/explicit_widgets.html
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply animations/explicit_widgets.md]: Are these the exceptions?
.html in: 
1. https://material.io/...  
2. https://codelabs.developers.google.com/...  
3. https://docs.flutter.io...

File: animations/transition_widgets.md (hunk 10 above)
Comment (czakdesign) [reply animations/transition_widgets.md]: remove 'complex'' but there are several occurrences in the hero and staggered animations doc that are not revised in this roll out.

File: animations/transition_widgets.md (hunk 11 above)
Comment (czakdesign) [reply animations/transition_widgets.md]: ! - text revised

File: animations/transition_widgets.md (hunk 12 above)
Comment (czakdesign) [reply animations/transition_widgets.md]: np!

File: animations/transition_widgets.md (hunk 14 above)
Comment (czakdesign) [reply animations/transition_widgets.md]: The implicit widgets that exist in the Flutter SDK must extent the ImplicityAnimatedWidget so that all the elements needed for an animation to work are included - it's an abstract class.

File: animations/transition_widgets.md (hunk 33 above)
Comment (czakdesign) [reply animations/transition_widgets.md]: Great -

**Comment by sfshaza2** [comment]
Track the animation docs in this PR:
[Rework animations docs #2690](https://github.com/flutter/website/issues/2690)

Closing.
